### PR TITLE
Bellatrix Jira Zephyr Plugin

### PR DESCRIPTION
**Pull request by MiriamKyoseva**

### CODE REVIEW

File: bellatrix.core/src/main/java/solutions/bellatrix/core/plugins/Plugin.java [hunk 1]
```diff
@@ -36,15 +36,32 @@ public void preBeforeTest(TestResult testResult, Method memberInfo) throws Excep
     public void postBeforeTest(TestResult testResult, Method memberInfo) {
     }
 
+
```
Comment (n1xan) [inline bellatrix.core/src/main/java/solutions/bellatrix/core/plugins/Plugin.java:39]: remove line

File: bellatrix.plugins.jira.zephyr/src/main/java/plugins/jira/zephyr/ZephyrPlugin.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    private String getExecutionId(Method memberInfo) {
+        if (memberInfo.isAnnotationPresent(ZephyrTestCase.class)) {
+            return memberInfo.getAnnotation(ZephyrTestCase.class).id();
+        } else {
+            return "";
+        }
+    }
+
+    private String getProjectId(Method memberInfo) {
+        if (memberInfo.getDeclaringClass().isAnnotationPresent(ZephyrProject.class)) {
```
Comment (n1xan) [inline bellatrix.plugins.jira.zephyr/src/main/java/plugins/jira/zephyr/ZephyrPlugin.java:75]: do we support the attribute only on the class level?

File: bellatrix.plugins.jira.zephyr/src/main/java/plugins/jira/zephyr/ZephyrPlugin.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            return memberInfo.getDeclaringClass().getAnnotation(ZephyrProject.class).id();
+        } else {
+            try {
+                return settings().getDefaultProjectKey();
+            } catch (Exception ex) {
+                return "";
```
[the remark this answers is not in the document]
Comment (n1xan) [reply bellatrix.plugins.jira.zephyr/src/main/java/plugins/jira/zephyr/ZephyrPlugin.java:81]: actually it is reading it from the config

File: bellatrix.plugins.jira.zephyr/src/main/java/plugins/jira/zephyr/ZephyrPlugin.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        }
+    }
+
+    @Override
+    public void postBeforeClass(Class type) {
+        if (!isEnabled()) return;
+
+        var testCycleName = String.format("%s %s",
+                DateTimeUtilities.getUtcNow(),
+                settings().getTestCycleName().isBlank() ? settings().getTestCycleName() : "BELLATRIX RUN");
```
Comment (n1xan) [inline bellatrix.plugins.jira.zephyr/src/main/java/plugins/jira/zephyr/ZephyrPlugin.java:100]: this is fine for now, but potentially we would want to set this via Environment variable.

File: bellatrix.plugins.jira.zephyr/src/main/java/plugins/jira/zephyr/ZephyrPlugin.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+import static io.restassured.RestAssured.given;
+
+public class ZephyrPlugin extends Plugin {
+    public static final EventListener<ZephyrCyclePluginEventArgs> ZEPHYR_CYCLE_CREATED = new EventListener<>();
+    public static final EventListener<ZephyrExecutionPluginEventArgs> ZEPHYR_TEST_CASE_EXECUTION_FAILED = new EventListener<>();
+    public static final EventListener<ZephyrExecutionPluginEventArgs> ZEPHYR_TEST_CASE_EXECUTED = new EventListener<>();
+    public static final EventListener<ZephyrCyclePluginEventArgs> ZEPHYR_CYCLE_STATUS_UPDATE_FAILED = new EventListener<>();
+
+    private ZephyrLocalData data;
```
Comment (n1xan) [inline bellatrix.plugins.jira.zephyr/src/main/java/plugins/jira/zephyr/ZephyrPlugin.java:44]: this is a bit generic - I would rename it to zaphyrData

File: bellatrix.plugins.jira.zephyr/src/main/java/plugins/jira/zephyr/ZephyrPlugin.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        data.getTestCycle().setPlannedEndDate(DateTimeUtilities.getUtcNow());
+
+        var response = ZephyrApiService.markTestCycleDone(data);
+
+        if (response.statusCode() >= 400) {
+            ZEPHYR_CYCLE_STATUS_UPDATE_FAILED.broadcast(new ZephyrCyclePluginEventArgs());
+        }
+    }
+
+    @Getter @Setter
```
Comment (n1xan) [inline bellatrix.plugins.jira.zephyr/src/main/java/plugins/jira/zephyr/ZephyrPlugin.java:145]: a class in the class? We usually create them in separate files

**Review by n1xan — CHANGES_REQUESTED** [review]
Please review and fix the PR comments.

**Review by teo-nikolov — APPROVED** [review]

**Pull request merged**